astropy/coordinates/baseframe.py

```diff
@@ -985,9 +985,9 @@ def __repr__(self):
 
         if self.has_data:
             if self.representation:
-                if (self.representation == SphericalRepresentation and
-                        isinstance(self.data, UnitSphericalRepresentation)):
-                    data = self.represent_as(UnitSphericalRepresentation,
+                if (issubclass(self.representation, SphericalRepresentation) and
+                    isinstance(self.data, UnitSphericalRepresentation)):
```

Comment:
Minor nitpick, don't bother fixing it, but when splitting an `if` statement` across multiple lines it's usually a bit nicer, visually, to indent the subsequent lines (as was this case in this line before you changed it).  Otherwise there's not much visual difference between the `if` statement itself and the code block under it.